src/explore-education-statistics-admin/src/components/TopLevelState.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { ReactNode, useState } from 'react';
+import TopLevelStateContext, {
+  SetThemeAndTopicsCallbackParameters,
+} from '@admin/components/TopLevelStateContext';
+import { IdTitlePair } from '@admin/services/common/types';
+import { ThemeAndTopics } from '@admin/services/dashboard/types';
+
+interface TopLevelStateProps {
+  children: ReactNode;
+}
+
+interface ThemeAndTopicsIdsAndTitles extends IdTitlePair {
+  topics: IdTitlePair[];
+}
+
+const themeToThemeWithIdTitleAndTopics = (theme: ThemeAndTopics) => ({
+  id: theme.id,
+  title: theme.title,
+  topics: theme.topics.map(topic => ({
+    id: topic.id,
+    title: topic.title,
+  })),
+});
+
+interface Model {
+  theme: ThemeAndTopicsIdsAndTitles;
+  topic: IdTitlePair;
+}
+
+const TopLevelState = ({ children }: TopLevelStateProps) => {
```

Comment:
The name is really generic
find and replace `TopLevelState` with `ThemeAndTopic`, maybe, something like that.

So we have `<ThemeAndTopicContext.Provider>` and `ThemeAndTopicProps` which are a bit more descriptive